lib/compat/wordpress-6.5/fonts/class-wp-font-utils.php

```diff
@@ -28,30 +28,26 @@ class WP_Font_Utils {
 		 * @link https://www.w3.org/TR/css-fonts-4/#font-family-prop
 		 *
 		 * @since 6.5.0
-		 * @access private
-		 *
-		 * @see sanitize_font_family()
 		 *
 		 * @param string $item A font family name.
-		 * @return string The font family name with surrounding quotes if necessary.
+		 * @return string The font family name with surrounding quotes, if necessary.
 		 */
 		private static function maybe_add_quotes( $item ) {
-			// Match any non alphabetic characters (a-zA-Z), dashes -, or parenthesis ().
-			$regex = '/[^a-zA-Z\-()]+/';
+			// Matches strings that are not exclusively alphabetic characters or hyphens, and do not exactly follow the pattern generic(alphabetic characters or hyphens).
+			$regex = '/^(?!generic\([a-zA-Z\-]+\)$)(?!^[a-zA-Z\-]+$).+/';
 			$item  = trim( $item );
 			if ( preg_match( $regex, $item ) ) {
-				// Removes leading and trailing quotes.
-				$item = preg_replace( '/^["\']|["\']$/', '', $item );
-				return "\"$item\"";
+				$item = trim( $item, "\"'" );
+				return '"' . $item . '"';
 			}
 			return $item;
 		}
 
 		/**
 		 * Sanitizes and formats font family names.
 		 *
-		 * - Applies `sanitize_text_field`
-		 * - Adds surrounding quotes to names that special
+		 * - Applies `sanitize_text_field`.
+		 * - Adds surrounding quotes to names that contain spaces or special characters.
```

Comment:
Looking at this again, I don't think digits are usually considered "special characters," so a comment something like the following seems clearer:

> Adds surrounding quotes to names containing any characters that are not alphabetic or dashes."

Reply:
done